filesystem/replicator_test.go

```diff
@@ -1,121 +1,141 @@
-			t.Fatal(err)
-		}
-		defer func() {
-			sourceFile.Close()
-			os.Remove("source_file.txt")
-		}()
-
-		sourceFilePath := "source_file.txt"
-		destinationFilePath := "destination_file.txt"
-		config := struct{}{}
-
-		err = replicateProcessFileCallBack(sourceFilePath, destinationFilePath, config)
-		if err != nil {
-			t.Errorf("Expected nil error, got: %s", err)
-		}
-	})
-
-	t.Run("CopySourceToDestinationDifferent", func(t *testing.T) {
-		sourceFile, err := os.Create("source_file.txt")
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer func() {
-			sourceFile.Close()
-			os.Remove("source_file.txt")
-		}()
-
-		_, err = sourceFile.WriteString("hello, world!")
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		sourceFile.Close() // Close and reopen the file to update the modification time
-		time.Sleep(1 * time.Second)
-
-		destinationFile, err := os.Create("destination_file.txt")
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer func() {
-			destinationFile.Close()
-			os.Remove("destination_file.txt")
-		}()
-
-		_, err = destinationFile.WriteString("hello, go!")
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		sourceFilePath := "source_file.txt"
-		destinationFilePath := "destination_file.txt"
-		config := struct{}{}
-
-		err = replicateProcessFileCallBack(sourceFilePath, destinationFilePath, config)
-		if err != nil {
-			t.Errorf("Expected nil error, got: %s", err)
-		}
-	})
-
-	t.Run("CopySourceToDestinationSame", func(t *testing.T) {
-		sourceFile, err := os.Create("source_file.txt")
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer func() {
-			sourceFile.Close()
-			os.Remove("source_file.txt")
-		}()
-
-		_, err = sourceFile.WriteString("hello, world!")
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		destinationFile, err := os.Create("destination_file.txt")
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer func() {
-			destinationFile.Close()
-			os.Remove("destination_file.txt")
-		}()
-
-		_, err = destinationFile.WriteString("hello, world!")
-		if err != nil {
-			t.Fatal(err)
-		}
-
-
-		sourceFilePath := "source_file.txt"
-		destinationFilePath := "destination_file.txt"
-		config := struct{}{}
-
-		err = replicateProcessFileCallBack(sourceFilePath, destinationFilePath, config)
-		if err != nil {
-			t.Errorf("Expected nil error, got: %s", err)
-		}
-	})
-}
+ 
+ func TestReplicateProcessFileCallBack(t *testing.T) {
+	 t.Run("test for source file does not exist, should return an error", func(t *testing.T) {
+		 sourceFilePath := "test_data/replicator_test/non_existent_file.txt"
+		 destinationFilePath := "test_data/replicator_test/destination.txt"
+		 config := struct{}{}
+		 err := replicateProcessFileCallBack(sourceFilePath, destinationFilePath, config)
+		 if err == nil {
+			 t.Errorf("Expected error for non-existent source file, got nil")
+		 }
+	 })
+ 
+	 t.Run("test for destination file doesn't exist, should copy the source file", func(t *testing.T) {
+		 sourceFilePath := "test_data/replicator_test/emptyfile.txt"
```

Comment:
I think the folder name has to be `testdata` https://github.com/golang/go/issues/14715
Also helps consistency https://github.com/konveyor/move2kube/tree/main/common/testdata

Reply:
There's one folder already present in the filesystem package with name "test_data". Shall I change it to "testdata"?

Reply:
https://github.com/konveyor/move2kube/tree/main/filesystem/test_data

Reply:
yes

Reply:
I have added the test files for this PR in "testdata" folder. Renaming the "test_data" folder will introduce changes in other files too, which may not be related to the exact purpose of this PR. Shall I can raise a separate PR for renaming it?

Reply:
@HarikrishnanBalagopal I have created the [PR ](https://github.com/konveyor/move2kube/pull/1072)for renaming the `test_data` folder to `testdata`